clash-lib/src/Clash/Core/FreeVars.hs

```diff
@@ -139,13 +116,6 @@ localVarsDoNotOccurIn
 localVarsDoNotOccurIn vs e =
   getAll (Lens.foldMapOf freeLocalVars (All . (`notElem` vs)) e)
 
--- | Check whether a local identifier occurs free in a term
-localIdOccursIn
-  :: Id
-  -> Term
-  -> Bool
-localIdOccursIn v e = getAny (Lens.foldMapOf freeLocalIds (Any . (== v)) e)
```

Comment:
I think it'd be worthwhile to use our benchmark suite to determine the performance effects of this change. As in, I feel the short-circuiting of `All`s monoidal merging, `&&`, would be faster than building up a varset using `unionVarSet` and then checking whether the element exists. But perhaps laziness saves us once again.